packages/media/src/webrtc/P2pRtcManager.ts

```diff
@@ -858,20 +874,24 @@ export default class P2pRtcManager implements RtcManager {
         this._fetchMediaServersTimer = null;
     }
 
-    _startMonitoringAudioTrack(track: any) {
+    _monitorAudioTrack(track: any) {
+        if (this._audioTrackBeingMonitored?.id === track.id) return;
+
+        this._audioTrackBeingMonitored && console.log("stop monitoring", this._audioTrackBeingMonitored.label);
+        console.log("start monitoring", track.label);
+        this._audioTrackBeingMonitored?.removeEventListener("ended", this._audioTrackOnEnded);
         track.addEventListener("ended", this._audioTrackOnEnded);
+        this._audioTrackBeingMonitored = track;
     }
 
-    _stopMonitoringAudioTrack(track: any) {
-        track.removeEventListener("ended", this._audioTrackOnEnded);
-    }
-    
-    _startMonitoringVideoTrack(track: CustomMediaStreamTrack) {
+    _monitorVideoTrack(track: CustomMediaStreamTrack) {
+        if (this._videoTrackBeingMonitored?.id === track.id) return;
+        
+        this._videoTrackBeingMonitored && console.log("stop monitoring", this._videoTrackBeingMonitored.label);
+        this._videoTrackBeingMonitored?.removeEventListener("ended", this._videoTrackOnEnded);
+        console.log("start monitoring", track.label);
```

Comment:
can we use some logger here or only console?

Reply:
It was only to make reviewing easier. Fixed in [a1aa382](https://github.com/whereby/sdk/pull/316/commits/a1aa38231e4745b1fb08dcf4b0c563a09e72dea6)